godot-core/src/task/futures.rs

```diff
@@ -375,8 +376,10 @@ pub unsafe trait DynamicSend: Send + Sealed {
 }
 
 /// Value that can be sent across threads, but only accessed on its original thread.
+///
+/// When moved to another thread, the inner value can no longer be accessed and will be leaked when the `ThreadConfined` is dropped.
 pub struct ThreadConfined<T> {
-    value: T,
+    value: Option<T>,
```

Comment:
We could do this just with a `ManuallyDrop` that we never actually drop, but then we couldn't implement `Drop` for `ThreadConfined` to log the error.

Reply:
I had `ManuallyDrop` first but it was messy to extract; the current one also needs no unsafe code.